src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -217,7 +217,14 @@ private void ExporterProc()
 
                 if (this.circularBuffer.Count > 0)
                 {
-                    this.exporter.Export(new Batch<T>(this.circularBuffer, this.maxExportBatchSize));
+                    try
+                    {
+                        this.exporter.Export(new Batch<T>(this.circularBuffer, this.maxExportBatchSize));
+                    }
+                    catch (Exception ex)
+                    {
+                        OpenTelemetrySdkEventSource.Log.SpanProcessorException(nameof(this.OnExport), ex);
```

Comment:
Do we need to drop the remaining items from the batch? Otherwise we might end up with a dead loop.

Add @CodeBlanch for awareness.

Reply:
@cijothomas @utpilla Were either of you able to reproduce the exception? The OtlpExporter has a try/catch over its comm so the throw would have to come from our code preparing the data?

@reyang Agree with you. This try/catch might swap an app crash for a deadlock. Depends on if the exporter was able to drain the buffer before the throw or not, but this code should be prepared for that case.

Reply:
> The OtlpExporter has a try/catch over its comm so the throw would have to come from our code preparing the data?

It doesn't catch general `Exception`, so need to fix it.

Reply:
Ah ya it catches `RpcException` we should change that to just `Exception`.